components/gitpod-db/src/typeorm/workspace-db-impl.ts

```diff
@@ -1032,6 +1032,24 @@ export abstract class AbstractTypeORMWorkspaceDBImpl implements WorkspaceDB {
         return await queryBuilder.getCount();
     }
 
+    public async getActiveUserCount(): Promise<number> {
+        const workspaceInstanceRepo = await this.getWorkspaceInstanceRepo();
+        const since = new Date();
+        since.setDate(since.getDate() - 7);
+
+        // get the owners of workspaces with new instances created at-least
+        // 3 days of the last 7 days
+        const queryBuilder = workspaceInstanceRepo
+            .createQueryBuilder("i")
+            .select("u.userId")
+            .where("i.startedTime  > :since", { since: since.toISOString() })
+            .innerJoinAndMapOne("i.user", DBWorkspaceInstanceUser, "u", "i.id = u.instanceId")
+            .groupBy("1")
+            .having("count(distinct date(i.startedTime)) >= 3");
```

Comment:
I think there's no need to join this over d_b_workspace_instance as `lastSeen` on `d_b_workspace_instance_usage` could be used.
E.g. something like:
```sql
SELECT count(distinct userid) from gitpod.d_b_workspace_instance_user where lastSeen> <since>
```

Reply:
@svenefftinge Hmm. So, Doing that would give us instances that are last seen > 7 days, but here we are looking for `users of workspaces with new instances created at-least 3 days of the last 7 days` which means we somehow need to relate the instance with `d_b_workspace_instance.startedTime` (to group by day and then count(unique days) >=3) and hence the join. Without `startedTime`, I'm not sure how we can do that? :thinking:

Reply:
Hmm, We might be able to use `lastSeen` itself instead of `startedTime` which means "users with workspaces instances that were last seen in at-least 3 of the 7 days`. Is that what you are suggesting? 🤔

Reply:
@Pothulapati This query looks veeeeery expensive. Can you validate the impact on DB CPU using [one of our read-only copies](https://www.notion.so/gitpod/Accessing-the-database-242e9fa0fef642128380a10d800d3b99#75a5c6647483496f8f085e43c8b9178d)? :thinking:

Reply:
This never runs in Saas, does it?

Reply:
Correct! The telemetry job and the setting page is disabled in SAAS. It's only aimed for self hosted instances.

Reply:
I've updated this query to be a lot more closer to what's in the Issue, and that works on BigQuery (takes around of 2 minutes there to execute but this won't be happening like mentioned above)

But I get the following error:

```
TypeError: Cannot read properties of undefined (reading 'getParameters')\n
```

Any idea why? 🤔